RubberduckTests/Refactoring/RenameFolder/RenameFolderRefactoringActionTests.cs

```diff
@@ -0,0 +1,202 @@
+            };
+
+            var refactoredCode = RefactoredCode(code, modelBuilder);
+
+            Assert.AreEqual(expectedCode, refactoredCode);
+        }
+
+        [Test]
+        [Category("Refactorings")]
+        public void RenameFolderRefactoringAction_ChangesExactlyTheSpecifiedModules()
+        {
+            const string code1 = @"
+'@Folder(""MyOldFolder.MyOldSubfolder.SubSub"")
+Public Sub Foo()
+End Sub
+";
+            const string code2 = @"
+'@Folder(""MyOldFolder.MyOldSubfolder"")
+Public Sub Foo()
+End Sub
+";
+            const string code3 = @"
+'@Folder(""MyOtherFolder.MyOldSubfolder"")
+Public Sub Foo()
+End Sub
+";
+            const string code4 = @"
+'@Folder(""MyOtherFolder.MyOtherSubfolder"")
+Public Sub Foo()
+End Sub
+";
+            const string code5 = @"
+Public Sub Foo()
+End Sub
+";
+            const string expectedCode1 = @"
+'@Folder ""MyOldFolder.MyNewFolder.SubSub""
+Public Sub Foo()
+End Sub
+";
+            const string expectedCode2 = @"
+'@Folder ""MyOldFolder.MyNewFolder""
+Public Sub Foo()
+End Sub
+";
+            const string expectedCode3 = code3;
+            const string expectedCode4 = code4;
+            const string expectedCode5 = code5;
+            Func<RubberduckParserState, RenameFolderModel> modelBuilder = (state) =>
+            {
+                var modules = state.DeclarationFinder
+                    .UserDeclarations(DeclarationType.Module)
+                    .OfType<ModuleDeclaration>()
+                    .ToList();
+
+                var module1 = modules.Single(module => module.IdentifierName.Equals("SubSubFolderModule"));
+                var module2 = modules.Single(module => module.IdentifierName.Equals("SubFolderModuleIncluded"));
+                const string originalFolder = "MyOldFolder.MyOldSubfolder";
+
+                return new RenameFolderModel(originalFolder, new List<ModuleDeclaration> { module1, module2 }, "MyNewFolder");
+            };
+
+            var refactoredCode = RefactoredCode(
+                    modelBuilder,
+                ("SubSubFolderModule", code1, ComponentType.StandardModule),
+                    ("SubFolderModuleIncluded", code2, ComponentType.ClassModule),
+                    ("SubFolderModuleNotIncluded", code3, ComponentType.ClassModule),
+                    ("UnaffectedSubFolderModule", code4, ComponentType.StandardModule),
+                    ("NoFolderModule", code5, ComponentType.StandardModule));
+
+            Assert.AreEqual(expectedCode1, refactoredCode["SubSubFolderModule"]);
+            Assert.AreEqual(expectedCode2, refactoredCode["SubFolderModuleIncluded"]);
+            Assert.AreEqual(expectedCode3, refactoredCode["SubFolderModuleNotIncluded"]);
+            Assert.AreEqual(expectedCode4, refactoredCode["UnaffectedSubFolderModule"]);
+            Assert.AreEqual(expectedCode5, refactoredCode["NoFolderModule"]);
```

Comment:
Nitpick: probably better off as `Assert.SequenceEquals` with an array containing the expected code, and another array containing the actual/refactored code.